src/main/java/de/dennisguse/opentracks/services/TrackRecordingManager.java

```diff
@@ -200,18 +219,17 @@ boolean onNewTrackPoint(@NonNull TrackPoint trackPoint) {
         if (distanceToLastStoredTrackPoint.greaterThan(maxRecordingDistance)) {
             trackPoint.setType(TrackPoint.Type.SEGMENT_START_AUTOMATIC);
             insertTrackPoint(trackPoint, true);
+
+            handler.removeCallbacks(ON_IDLE);
+            handler.postDelayed(ON_IDLE, idleDuration.toMillis());
             return true;
         }
 
-        if (distanceToLastStoredTrackPoint.greaterOrEqualThan(recordingDistanceInterval)
-                && trackPoint.isMoving()) {
+        if (distanceToLastStoredTrackPoint.greaterOrEqualThan(recordingDistanceInterval)) {
             insertTrackPoint(trackPoint, false);
-            return true;
-        }
 
-        if (trackPoint.isMoving() != lastStoredTrackPoint.isMoving()) {
-            // Moving from non-moving to moving or vice versa; required to compute moving time correctly.
-            insertTrackPoint(trackPoint, true);
+            handler.removeCallbacks(ON_IDLE);
+            handler.postDelayed(this::onIdle, idleDuration.toMillis());
```

Comment:
Same here: should `this::onIdle` be `ON_IDLE`?

Reply:
yes